package/src/utils/utils.ts

```diff
@@ -234,32 +229,30 @@ const queryUsers = async <
     mentionAllAppUsersQuery?: MentionAllAppUsersQuery<StreamChatGenerics>;
   } = {},
 ): Promise<void> => {
-  if (typeof query === 'string') {
+  if (!query) return;
+  try {
     const {
       limit = defaultAutoCompleteSuggestionsLimit,
       mentionAllAppUsersQuery = defaultMentionAllAppUsersQuery,
     } = options;
+
     const filters = {
-      id: { $ne: client.userID },
+      $or: [{ id: { $autocomplete: query } }, { name: { $autocomplete: query } }],
       ...mentionAllAppUsersQuery?.filters,
     };
 
-    if (query) {
-      // @ts-ignore
-      filters.$or = [{ id: { $autocomplete: query } }, { name: { $autocomplete: query } }];
-    }
-
-    const response = await client.queryUsers(
+    const { users } = await client.queryUsers(
       // @ts-ignore
       filters,
       { id: 1, ...mentionAllAppUsersQuery?.sort },
       { limit, ...mentionAllAppUsersQuery?.options },
     );
-    const users: SuggestionUser<StreamChatGenerics>[] = [];
-    response.users.forEach((user) => isUserResponse(user) && users.push(user));
+    const usersWithoutClientUserId = users.filter((user) => user.id !== client.userID);
     if (onReady && users) {
-      onReady(users);
+      onReady(usersWithoutClientUserId);
     }
+  } catch (error) {
+    console.warn('Error querying users:', error);
```

Comment:
Shouldn't we bubble this error up? How can one recover from an error that can happen here?